### DISCUSSION TITLE
how to publish localnetwork rtsp stream to public IP?

### DISCUSSION THREAD

**Discussion by solfKwolf** [bluenviron/mediamtx#1493] [Questions]
I start up a rtsp-simple-server with a pubic ip,and want to publish localnetwork rtsp stream to this public IP,but it's failed.
![image](https://user-images.githubusercontent.com/30661539/220071064-b10cbeb3-cd19-48f6-ac42-6183398ff996.png)

**Comment by dbuscaino** [comment]
Hi, can you give more information such as your configuration files and how you publish the stream?

From the image you published I see that you try to send the stream to the port 23713 of the machine with public ip:
- Is the port correct? 
- Did you change the rtsp listener port in the rtsp-simple-server config file on the public ip machine? 
- Do you have a firewall? 
- Did you open the port 23713 on the public machine? 
- Are you able to send the stream directly to the machine with public ip?

**Reply by rpatidar** [reply]
> I start up a rtsp-simple-server with a pubic ip,and want to publish localnetwork rtsp stream to this public IP,but it's failed. ![image](https://user-images.githubusercontent.com/30661539/220071064-b10cbeb3-cd19-48f6-ac42-6183398ff996.png)

see if this works ?
https://github.com/aler9/rtsp-simple-server/discussions/1529#discussioncomment-5208391

> in the config file changing  `rtspAddress: :8554` to `rtspAddress: 0.0.0.0:8554`  works ?

**Comment by aler9** [comment]
Hello, if you're seeing the `[conn x.x.x.x] opened` line it means that your client is able to connect to the server though the public IP and there are no firewall issues.
Make sure that the client is using the right protocol (RTSP for port 8554).

[a comment by an automated account is not shown]
